wdl2cwl/expr.py

```diff
@@ -0,0 +1,288 @@
+        if isinstance(
+            part,
+            (WDL.Expr.Placeholder, WDL.Expr.Apply, WDL.Expr.Get, WDL.Expr.Ident),
+        ):
+            placeholder = get_expr(part, ctx)
+            part = (
+                "" if parts[index - 1] == '"' or parts[index - 1] == "'" else "' + "  # type: ignore
+            )
+            part += placeholder
+            part += (
+                "" if parts[index + 1] == '"' or parts[index + 1] == "'" else " + '"  # type: ignore
+            )
+        string += part
+    # condition to determine if the opening and closing quotes should be added to string
+    # for cases where a placeholder begins or ends a WDL.Expr.String
+    if type(parts[1]) == str:
+        string = "'" + string
+    if type(parts[-2]) == str:
+        string = string + "'"
+    return string
+
+
+def get_expr_name(wdl_expr: WDL.Expr.Ident) -> str:
+    """Extract name from WDL expr."""
+    if not hasattr(wdl_expr, "name"):
+        raise WDLSourceLine(wdl_expr, ConversionException).makeError(
+            f"{type(wdl_expr)} has not attribute 'name'"
+        )
+    return get_input(wdl_expr.name)
+
+
+def get_expr(wdl_expr: Any, ctx: ConversionContext) -> str:
+    """Translate WDL Expressions."""
+    if isinstance(wdl_expr, WDL.Expr.Apply):
+        return get_expr_apply(wdl_expr, ctx)
+    elif isinstance(wdl_expr, WDL.Expr.Get):
+        return get_expr_get(wdl_expr, ctx)
+    elif isinstance(wdl_expr, WDL.Expr.IfThenElse):
+        return get_expr_ifthenelse(wdl_expr, ctx)
+    elif isinstance(wdl_expr, WDL.Expr.Placeholder):
+        return translate_wdl_placeholder(wdl_expr, ctx)
+    elif isinstance(wdl_expr, WDL.Expr.String):
+        return get_expr_string(wdl_expr, ctx)
+    elif isinstance(wdl_expr, WDL.Tree.Decl):
+        return get_expr(wdl_expr.expr, ctx)
+    elif isinstance(
+        wdl_expr,
+        (
+            WDL.Expr.Boolean,
+            WDL.Expr.Int,
+            WDL.Expr.Float,
+            WDL.Expr.Array,
+        ),
+    ):
+        return get_literal_name(wdl_expr)
+    else:
+        raise WDLSourceLine(wdl_expr, ConversionException).makeError(
+            f"The expression '{wdl_expr}' is not handled yet."
+        )
+
+
+def get_expr_apply(wdl_apply_expr: WDL.Expr.Apply, ctx: ConversionContext) -> str:
+    """Translate WDL Apply Expressions."""
+    function_name = wdl_apply_expr.function_name
+    arguments = wdl_apply_expr.arguments
+    if not arguments:
+        raise WDLSourceLine(wdl_apply_expr, ConversionException).makeError(
+            f"The '{wdl_apply_expr}' expression has no arguments."
+        )
+    treat_as_optional = wdl_apply_expr.type.optional
+
+    # Call the function if we have it in our wdl2cwl.functions module
+    if hasattr(functions, function_name):
+        return cast(
+            str,
+            getattr(functions, function_name)(
+                arguments, ctx, treat_as_optional=treat_as_optional
+            ),
+        )
```

Comment:
The rest of the code in this module should be very similar (if not exact copy) of the code from `main.py`. This `_apply` function was modified to dynamically call a WDL Stdlib by its name from `wdl2cwl.functions`.